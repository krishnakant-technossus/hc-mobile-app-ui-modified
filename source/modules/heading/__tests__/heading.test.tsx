// npx jest ./source/modules/heading/__tests__/heading.test.js

import React from 'react';
import {render} from '@testing-library/react-native';

import Heading from '../heading';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('Heading', () => {
  it('renders itself correctly', () => {
    const {getByTestId} = render(
      <ThemeProvider theme={defaultTheme}>
        <Heading testID="simpleHeading">This is a heading</Heading>
      </ThemeProvider>,
    );

    const textElement = getByTestId('simpleHeading');

    expect(textElement).toBeDefined();
  });

  it('renders renders the text correctly', () => {
    const {getByText} = render(
      <ThemeProvider theme={defaultTheme}>
        <Heading testID="simpleHeading">This is a heading</Heading>
      </ThemeProvider>,
    );

    const textElement = getByText('This is a heading');

    expect(textElement).toBeDefined();
  });
});
