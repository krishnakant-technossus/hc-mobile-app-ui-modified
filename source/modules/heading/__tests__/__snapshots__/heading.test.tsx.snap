// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Rendering Variants alignment-center: should match snapshot 1`] = `
<Text
  alignment="center"
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 24,
        "textAlign": "center",
      },
    ]
  }
  testID="heading"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering Variants alignment-left: should match snapshot 1`] = `
<Text
  alignment="left"
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 24,
        "textAlign": "left",
      },
    ]
  }
  testID="heading"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering Variants alignment-right: should match snapshot 1`] = `
<Text
  alignment="right"
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 24,
        "textAlign": "right",
      },
    ]
  }
  testID="heading"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering Variants h1: should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 24,
        "textAlign": "left",
      },
    ]
  }
  testID="heading"
  variant="h1"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering Variants h2: should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 20,
        "textAlign": "left",
      },
    ]
  }
  testID="heading"
  variant="h2"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering Variants h3: should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 18,
        "textAlign": "left",
      },
    ]
  }
  testID="heading"
  variant="h3"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering Variants h4: should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 16,
        "textAlign": "left",
      },
    ]
  }
  testID="heading"
  variant="h4"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering Variants h5: should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 16,
        "textAlign": "left",
      },
    ]
  }
  testID="heading"
  variant="h5"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering Variants h6: should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 14,
        "textAlign": "left",
      },
    ]
  }
  testID="heading"
  variant="h6"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering Variants inversed: should match snapshot 1`] = `
<Text
  inversed={true}
  style={
    Array [
      Object {
        "color": "#FFFFFF",
        "fontFamily": "System",
        "fontSize": 24,
        "textAlign": "left",
      },
    ]
  }
  testID="heading"
>
  This is a heading
</Text>
`;

exports[`Heading Rendering renders itself correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1C2530",
        "fontFamily": "System",
        "fontSize": 24,
        "textAlign": "left",
      },
    ]
  }
  testID="heading"
>
  This is a heading
</Text>
`;
